### ISSUE TITLE
[CRASH] - OpenSIPS Crashes after call remove_ip_port function 

### ISSUE THREAD

**Issue by eliakimfelmorais** [OpenSIPS/opensips#2954]
**OpenSIPS 3.1**

**Version:** 
```
version: opensips 3.1.3 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 50fae83
main.c compiled on 00:06:05 Jul 14 2021 with gcc 4.8.5
```

**Crash Core Dump**

https://drive.google.com/file/d/1f8jeVOWg93HjW6BIWywWCVpcrWwmdJPl/view?usp=sharing

This problem happened after we call the function **remove_ip_port** from **registrar** module. This function is triggered by the route bellow:
```
event_route[E_CORE_TCP_DISCONNECT] {
    if (is_myself($param(dst_ip),$param(dst_port))) {
        xlog("L_WARN", "TCP disconnection from $param(src_ip):$param(src_port) type $param(proto). Removing contact from the location table\n");
        remove_ip_port($param(src_ip),$param(src_port), "location");
    } else {
        xlog("L_INFO", "Outbound TCP disconnection to $param(src_ip):$param(src_port) type $param(proto)\n");
    }
}
```
After I had some network issue this event is generated internally to disconnect some extensions with broken connections. So after receive the **E_CORE_TCP_DISCONNECT** we remove the contacts using **remove_ip_port**. In the backtrace file is possible to see the behavior. 

I tried to look the map.c its look a conversion to integer is going wrong. I got the pointers used to look if some parameter is wrong. 
```
$1 = {s = 0x7fdc8a6aec3b "10.48.38.248src_portdst_ip10.48.22.3dst_portprotowss", len = 12}
(gdb) p *port
$2 = 60024
```
The parameters in the function are right.

After this error the OpenSIPS crashes and starts again with systemd.

Please tell me if I can provide more information about this issue.

**OS/environment information**
 cat /etc/redhat-release
CentOS Linux release 7.8.2003 (Core)
Linux EQTLTSPSEDSACWBP 3.10.0-1127.19.1.el7.x86_64 #1 SMP Tue Aug 25 17:23:54 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Thank you.

**Comment by bogdan-iancu** [comment]
@eliakimfelmorais , could you please extract and post the full BT from the corefile (run `bt full` in gdb)

**Comment by eliakimfelmorais** [comment]
Yes! I appended the bt full 

[bt_full_crash.txt](https://github.com/OpenSIPS/opensips/files/10031299/bt_full_crash.txt)

**Comment by eliakimfelmorais** [comment]
Right! I made the update to version opensips 3.3.2 and I'll make more tests.

This function was developed as a custom change by Vlad Paiu.  Is there any way if we test in 3.3 version and if we get same error hire a custom support of OpenSIPS Solutions team?

Thank you Bogdan!

**Comment by eliakimfelmorais** [comment]
Hi Bogdan. How you doing ? 

The error occurred again. 

[root@sippulse-hpbx opensips]# opensips -V
version: opensips 3.3.2 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 6620aa5
main.c compiled on 21:41:18 Nov 22 2022 with gcc 4.8.5


[gdb_balancer_remove_ip_port.txt ](https://github.com/OpenSIPS/opensips/files/10085005/gdb_balancer_remove_ip_port.txt)

**Comment by eliakimfelmorais** [comment]
Should I check if the IP is registered before make the remove ? 

I checked the registrar module has the function 

https://opensips.org/docs/modules/3.3.x/registrar.html#func_is_ip_registered

But  is_ip_registered does not execute on event_route? Can I modify to test to execute in event_route?

**Comment by bogdan-iancu** [comment]
@eliakimfelmorais , what clustering / preset / db_mode are you using in usrloc module ?
Also, can we arrange direct access to a server where I can inspect the corefile? If so please ping me off-ticket.

**Comment by eliakimfelmorais** [comment]
Actually We are using 

```
modparam("usrloc", "working_mode_preset", "single-instance-sql-write-through"). 
```
How can we schedule the time and the access? Where Can I ping you ?

Thank you.

**Comment by liviuchircu** [comment]
@eliakimfelmorais:  This bug should be fixed now, there was a possible invalid memory access after deleting the AoR, which matches your experience of getting random crashes after calling `remove_ip_port()`.  Let me know if you run into any more issues here!

PS: this fix will be included in the upcoming `3.3.4` release, due Feb 22nd.
